src/pudl/etl/ferceqr_deployment.py

```diff
@@ -0,0 +1,145 @@
+        logger.info(f"Copying {table} to GCS.")
+        base_path = UPath(output_location) / table
+        base_path.mkdir(exist_ok=True)
+
+        # Loop through partitioned parquet files for table and write to GCS
+        for file in ParquetData(table_name=table).parquet_directory.iterdir():
+            destination_path = base_path / file.name
+            destination_path.write_bytes(file.read_bytes())
+
+    # Send slack notification about successful build
+    logger.info("Notifying slack about successful build.")
+    _notify_slack_deployments_channel(
+        ":large_green_circle: :sunglasses: :unicorn_face: :rainbow: ferceqr deployment succeeded!!"
+        " :partygritty: :database_parrot: :blob-dance: :large_green_circle:\n\n"
+        f"Parquet files can be found at: {output_location}"
+    )
+    _write_status_file("SUCCESS")
+
+
+@dg.asset
+def handle_ferceqr_deployment_failure():
+    """Send notification if EQR deployment failed."""
+    logger.info("Build failed, notifying slack.")
+    _notify_slack_deployments_channel(
+        message=":x: ferceqr deployment failed! See step status here:",
+        attached_file_path=str(_get_etl_status_csv_path()),
+    )
+    _write_status_file("FAILURE")
+
+
+def _get_etl_status_csv_path() -> Path:
+    return PudlPaths().output_dir / "ferceqr_etl_status.csv"
+
+
+@dg.sensor(
+    default_status=ferceqr_sensor_status,
+    minimum_interval_seconds=60,
+    asset_selection=["deploy_ferceqr", "handle_ferceqr_deployment_failure"],
+)
+def ferceqr_sensor(context: dg.RunStatusSensorContext):
+    """Check if the EQR backfill is complete and handle appropriately."""
+    asset_statuses = {}
+
+    # Query status of all assets across all partitions
+    for asset_key in FERCEQR_TRANSFORM_ASSETS + ["extract_eqr"]:
+        partition_statuses = context.instance.get_status_by_partition(
+            asset_key=dg.AssetKey(asset_key),
+            partition_keys=ferceqr_year_quarters.get_partition_keys(),
+            partitions_def=ferceqr_year_quarters,
+        )
+        asset_statuses[asset_key] = partition_statuses
+
+    # Check if all partitions have been attempted
+    asset_statuses = pd.DataFrame(asset_statuses)
+    in_progress_parts = (asset_statuses == dg.AssetPartitionStatus.IN_PROGRESS).any(
+        axis=1
+    )
+    not_started_parts = asset_statuses[FERCEQR_TRANSFORM_ASSETS].isnull().any(
+        axis=1
+    ) & (asset_statuses["extract_eqr"] != dg.AssetPartitionStatus.FAILED)
+
+    # Don't do anything if there are still partitions that haven't finished running
+    if (in_progress_parts | not_started_parts).any():
+        logger.info("Partitions still in progress, continuing.")
+        return None
+    # The backfill must be complete to reach this point
+    # Check if any partitions failed
+    if (asset_statuses == dg.AssetPartitionStatus.FAILED).any(axis=1).any():
+        # Write status to CSV file
+        asset_statuses.to_csv(_get_etl_status_csv_path())
```

Comment:
Nice, this seems like it was super useful in debugging :)

Reply:
Hopefully! At first I was considering sending an a formatted table in the slack message, but decided it wasn't worth the hassle.